corehq/apps/reports/standard/deployments.py

```diff
@@ -663,6 +668,55 @@ class AggregateUserStatusReport(ProjectReport, ProjectReportParametersMixin):
     def decorator_dispatcher(self, request, *args, **kwargs):
         super(AggregateUserStatusReport, self).decorator_dispatcher(request, *args, **kwargs)
 
+    @property
+    @memoized
+    def selected_app_id(self):
+        return self.request_params.get(SelectApplicationFilter.slug, None)
+
+    def _get_histogram_aggregation_for_app(self, nested_field_name, date_field_name):
+        """
+        The histogram aggregation is put inside a nested and filter aggregation to only query
+        the nested documents that match the selected app ID.
+
+        This function will return a nested aggregation that looks like the following:
+        "aggs": {
+            "nested_{nested_field_name}": {
+                "nested": {
+                    "path": "reporting_metadata.{nested_field_name}"
+                }
+            },
+            "aggs": {
+                "filtered_{nested_field_name}": {
+                    "filter": {
+                        "term": {
+                            "reporting_metadata.{nested_field_name}.app_id": self.selected_app_id
+                        }
+                    }
+                    "aggs": {
+                        "{nested_field_name}_date_histogram": {
+                            "date_histogram": {
+                                ...
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        """
+        nested_field_path = f'reporting_metadata.{nested_field_name}'
```

Comment:
nit: same note here about naming. I believe it would be better to just call it "field_path"

Reply:
[9aaabbf](https://github.com/dimagi/commcare-hq/pull/34178/commits/9aaabbfb5ae31c4d87cc4f20fc293b69026b7043)